test/pytest/test_peer.py

```diff
@@ -189,7 +189,14 @@ def __init__(self, *, specific_error=False):
         async def do_connect_transport(self) -> None:
             transport = await self.spawn(['sh', '-c', 'exit 9'], ())
             assert isinstance(transport, SubprocessTransport)
-            time.sleep(0.1)  # sync! increase chance for process to end already
+            # wait until process is dead without calling wait() on it, as that would interfere
+            # with the transport's child watching
+            pid = transport.get_pid()
+            while True:
+                with open(f'/proc/{pid}/stat') as f:
+                    if f.read().split()[2] == 'Z':
+                        break
+                time.sleep(0.1)
```

Comment:
I'm generally okay with this approach, but you could also do something like:

```python
# The process will exit soon — try writing to it until a write fails.
while not transport.is_closing():
    transport.write(b'x')
    time.sleep(0.1)
```

no?

Reply:
Ah, nice idea -- catch it from the write side instead of the process exit side. Seems to work fine!